crypto/dh/dh_lib.c

```diff
@@ -235,11 +235,11 @@ void DH_get0_key(const DH *dh, const BIGNUM **pub_key, const BIGNUM **priv_key)
 int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key)
 {
     if (pub_key != NULL) {
-        BN_free(dh->pub_key);
+        BN_clear_free(dh->pub_key);
```

Comment:
Does `pub_key` really need to be cleared?

Reply:
No, but it is clear free'd in DH_free together with even more public
things like g, p, q, etc. but I would not like to change DH_free to use BN_free.
So this is for consistency.

Reply:
It's okay, I was just querying the change which seemed odd.  I'm happy to clear and free this, it is always safer.

Reply:
It seems strange to me that DH_free does that for public fields

Reply:
Yes. me too.